pandas/core/arrays/datetimelike.py

```diff
@@ -140,6 +139,67 @@ def asi8(self):
         # do not cache or you'll create a memory leak
         return self.values.view('i8')
 
+    # ------------------------------------------------------------------
+    # Extension Array Interface
+    # TODO:
+    #   _from_sequence
+    #   _from_factorized
+    #   __setitem__
+    #   _values_for_argsort
+    #   argsort
+    #   fillna
+    #   dropna
+    #   shift
+    #   unique
+    #   _values_for_factorize
+    #   factorize
+    #   _formatting_values
+    #   _reduce
+
+    def _validate_fill_value(self, fill_value):
+        """
+        If a fill_value is passed to `take` convert it to an i8 representation,
+        raising ValueError if this is not possible.
+
+        Parameters
+        ----------
+        fill_value : object
+
+        Returns
+        -------
+        fill_value : np.int64
+
+        Raises
+        ------
+        ValueError
+        """
+        raise AbstractMethodError(self)
+
+    def take(self, indices, allow_fill=False, fill_value=None):
+
+        if allow_fill:
+            fill_value = self._validate_fill_value(fill_value)
+
+        new_values = take(self._data,
+                          indices,
+                          allow_fill=allow_fill,
+                          fill_value=fill_value)
+
+        return self._shallow_copy(new_values)
+
+    @classmethod
+    def _concat_same_type(cls, to_concat):
```

Comment:
This is not really tested for now, is that correct?

Reply:
Correct.

Reply:
Sorry, but I am -1 on moving things out of the PeriodArray PR that cannot stand on its own

Reply:
Assuming you're implicitly saying you'd be OK with this if tests were implemented as part of this PR, I think this is an excellent comment.

Reply:
No, that is not really what I meant :-) 
Because now you added tests for something that would already be tested in the PeriodArray PR. We don't need explicit tests for this method, as it is tested through all existing methods for concat, and it already has explicit tests in the base extension tests.

So I just think this simply belongs in a PR where we actually make PeriodArray an ExtensionArray.

Reply:
> that would already be tested in the PeriodArray PR. 

The benefit of doing things in smaller chunks is that each chunk gets more focused.  e.g. I don't want to throw shade on Tom's PR, but the `take` implementation missed `fill_value` that is a `Period` with non-matching `freq`.  Doing a couple methods at a time lets both authors and reviewers focus more precisely.

And while "too many tests" does have a downside, it really isn't in the top 5 things I'm going to worry about today.